### ISSUE TITLE
Github password is stored as plain text in axoloti.prefs

### ISSUE THREAD

**Issue by vormplus** [axoloti/axoloti#322]
I know this feature is to make it handy for people to just press the sync library button to upload the objects they just made to the community library, but is it possible to use certificates instead?

**Comment by TheTechnobear** [comment]
yeah, i know.. and honestly, I knew someone would ask for it to change at some point :) 

certs are a possibility, and don't change to much of the code for the actual repo interaction.
I suspect the most of the effort is making the whole certificate creation and assigning to github account, painless and non-technical.

I think, Id be tempted to do down a route of providing alternative authentication mechanism, e.g. leave simple passwords, for where users are happy with this.
 
the certificate may also be useful for other things we want to do with axoloti too e.g. packaging/signing.

note: a 'quick' implementation may actually be to allow use of a cert if it exists, i.e. for technical users who are unhappy with the password auth, so are familiar with certificates, and have their own already.
I think this is not much more than using the certificate during the initial clone, but we would need to check there are not corner cases in the code..
